build-logic/project-plugins/src/main/kotlin/com.hedera.hashgraph.maven-publish.gradle.kts

```diff
@@ -25,54 +25,50 @@ java {
-            versionMapping {
-                usage("java-api") { fromResolutionResult() }
-                usage("java-runtime") { fromResolutionResult() }
-            }
-
-            pom {
-                packaging = findProperty("maven.project.packaging")?.toString() ?: "jar"
-                name.set(project.name)
-                url.set("https://www.swirlds.com/")
-                inceptionYear.set("2016")
+val maven =
+    publishing.publications.create<MavenPublication>("maven") {
+        from(components.getByName("java"))
+        versionMapping {
+            usage("java-api") { fromResolutionResult() }
+            usage("java-runtime") { fromResolutionResult() }
+        }
 
-                description.set(provider(project::getDescription))
+        pom {
+            packaging = findProperty("maven.project.packaging")?.toString() ?: "jar"
+            name.set(project.name)
+            url.set("https://www.swirlds.com/")
+            inceptionYear.set("2016")
 
-                organization {
-                    name.set("Hedera Hashgraph, LLC")
-                    url.set("https://www.hedera.com")
-                }
+            description.set(provider(project::getDescription))
 
-                licenses {
-                    license {
-                        name.set("Apache License, Version 2.0")
-                        url.set(
-                            "https://raw.githubusercontent.com/hashgraph/hedera-services/main/LICENSE"
-                        )
-                    }
-                }
+            organization {
+                name.set("Hedera Hashgraph, LLC")
+                url.set("https://www.hedera.com")
+            }
 
-                scm {
-                    connection.set("scm:git:git://github.com/hashgraph/hedera-services.git")
-                    developerConnection.set(
-                        "scm:git:ssh://github.com:hashgraph/hedera-services.git"
+            licenses {
+                license {
+                    name.set("Apache License, Version 2.0")
+                    url.set(
+                        "https://raw.githubusercontent.com/hashgraph/hedera-services/main/LICENSE"
                     )
-                    url.set("https://github.com/hashgraph/hedera-services")
                 }
             }
+
+            scm {
+                connection.set("scm:git:git://github.com/hashgraph/hedera-services.git")
+                developerConnection.set("scm:git:ssh://github.com:hashgraph/hedera-services.git")
+                url.set("https://github.com/hashgraph/hedera-services")
+            }
         }
     }
-}
 
-signing {
-    useGpgCmd()
-    sign(publishing.publications.getByName("maven"))
-}
+val publishSigningEnabled =
+    providers.gradleProperty("publishSigningEnabled").getOrElse("false").toBoolean()
 
-tasks.withType<Sign>().configureEach {
-    onlyIf { providers.gradleProperty("publishSigningEnabled").getOrElse("false").toBoolean() }
```

Comment:
Accessing `gradleProperty` inside `onlyIf` is not allowed (`onlyIf` runs "too close" to execution time).
Instead, we make the whole "sigining {}" configuration conditional (see below).